example/example.md

```diff
@@ -0,0 +1,601 @@
+#### Listen for all realtime presence messages
+
+```dart
+StreamSubscription<ably.PresenceMessage> subscription =
+    channel
+        .presence
+        .subscribe()
+        .listen((presenceMessage) {
+            // Handle presence message
+        },
+);
+```
+
+#### Listen for a particular presence message
+
+```dart
+StreamSubscription<ably.PresenceMessage> subscription =
+    channel
+        .presence
+        .subscribe(action: PresenceAction.enter)
+        .listen((presenceMessage) {
+            // Handle `enter` presence message
+        },
+);
+```
+
+#### Listen for a set of particular presence messages
+
+```dart
+StreamSubscription<ably.PresenceMessage> subscription =
+    channel
+        .presence
+        .subscribe(actions: [
+            PresenceAction.enter,
+            PresenceAction.update,
+        ])
+        .listen((presenceMessage) {
+            // Handle `enter` and `update` presence message
+        },
+);
+```
+
+## Using the REST API
+
+### Rest instance
+
+#### Create instance of REST API
+
+```dart
+ably.Rest rest = ably.Rest(options: clientOptions);
+```
+
+#### Read Rest time
+
+```dart
+DateTime time = rest.time()
+```
+
+### Rest channel
+
+#### Create instance of Rest channel
+
+```dart
+ably.RestChannel channel = rest.channels.get('channel-name');
+```
+
+### Rest channel messages
+
+#### Publish single message on Rest channel
+
+```dart
+// Publish simple message
+await channel.publish(
+    name: "event1",
+    data: "hello world",
+);
+
+// Publish message data as json-encodable object
+await channel.publish(
+    name: "event1",
+    data: {
+        "hello": "world",
+        "hey": "ably",
+    },
+);
+
+// Publish message as array of json-encodable objects
+await channel.publish(
+    name: "event1",
+    data: [
+        {
+        "hello": {
+            "world": true,
+        },
+        "ably": {
+            "serious": "realtime",
+        },
+    ],
+);
+
+// Publish message as an `ably.Message` object
+await channel.publish(
+    message: ably.Message(
+        name: "event1",
+        data: {
+            "hello": "world",
+        }
+    ),
+);
+```
+
+#### Publish multiple messages on Rest channel
+
+```dart
+await channel.publish(
+    messages: [
+        ably.Message(
+            name: "event1",
+            data: {
+                "hello": "world",
+            }
+        ),
+        ably.Message(
+            name: "event1",
+            data: {
+                "hello": "ably",
+            }
+        ),
+    ],
+);
+```
+
+### Rest channel history
+
+#### Read Rest channel history
+
+```dart
+// Get channel history with default parameters
+ably.PaginatedResult<ably.Message> history = await channel.history()
+
+// Get channel history with custom parameters
+ably.PaginatedResult<ably.Message> filteredHistory = await channel.history(
+    ably.RestHistoryParams(
+        direction: 'forwards',
+        limit: 10,
+    )
+)
+```
+
+### Rest presence
+
+#### Enter Rest presence
```

Comment:
Is it even allowed to enter the presence of a Rest channel? Looking into the spec I've found [RSP2](https://ably.com/docs/client-lib-development-guide/features#RSP2) which states:
> There is no way to register a member as present on a channel via the REST API

So it might mean that there is no enter, update and leave operations on a Rest presence channel :thinking: @QuintinWillison WDYT?

Reply:
Good catch, it's probably a copy-paste error, these shouldn't be included for Rest. My bad, I'll remove them and leave only the history method

Reply:
Fixed in https://github.com/ably/ably-flutter/pull/384/commits/24e4359a9f3b35dd2ed576e9f4a3e9222acf7b73